ZombieSharp/EventModule.cs

```diff
@@ -76,7 +76,11 @@ private HookResult OnPlayerConnectFull(EventPlayerConnectFull @event, GameEventI
 
             Logger.LogInformation($"Player: {client.PlayerName} is fully connected with {client.Slot}");
 
-            PlayerSettingsAuthorized(client).Wait();
+            var AuthTask = PlayerSettingsAuthorized(client);
+            var StatTask = StatsSetData(client);
+
+            Task.WhenAll(AuthTask, StatTask).Wait();
```

Comment:
Do not mix sync and async code together. Is it possible to make everything asynchronous?
By using async Tasks you are adding a bit overhead for adding and support internal Task state machine which gives you benefits with async code. But here you use `.Wait()`, so you are losing these benefits and only adding overhead

Reply:
I ran into trouble with non-main thread sometime.

Reply:
> I ran into trouble with non-main thread sometime.

Does CounterStrikeSharp have native support for register async event handlers? If no, then alright...

Reply:
they don't